core/service/manager.go

```diff
@@ -18,47 +18,64 @@
 package service
 
 import (
-	"crypto/x509/pkix"
-
 	log "github.com/cihub/seelog"
-	"github.com/mysteriumnetwork/go-openvpn/openvpn"
-	"github.com/mysteriumnetwork/go-openvpn/openvpn/middlewares/state"
-	"github.com/mysteriumnetwork/go-openvpn/openvpn/tls"
 	"github.com/mysteriumnetwork/node/communication"
-	"github.com/mysteriumnetwork/node/core/ip"
-	"github.com/mysteriumnetwork/node/core/location"
+	nats_dialog "github.com/mysteriumnetwork/node/communication/nats/dialog"
+	nats_discovery "github.com/mysteriumnetwork/node/communication/nats/discovery"
 	"github.com/mysteriumnetwork/node/discovery"
 	"github.com/mysteriumnetwork/node/identity"
+	identity_registry "github.com/mysteriumnetwork/node/identity/registry"
 	identity_selector "github.com/mysteriumnetwork/node/identity/selector"
+	"github.com/mysteriumnetwork/node/logconfig"
 	"github.com/mysteriumnetwork/node/metadata"
-	"github.com/mysteriumnetwork/node/nat"
 	dto_discovery "github.com/mysteriumnetwork/node/service_discovery/dto"
-	openvpn_discovery "github.com/mysteriumnetwork/node/services/openvpn/discovery"
-	"github.com/mysteriumnetwork/node/session"
 )
 
-// Manager represent entrypoint for Mysterium service with top level components
+const logPrefix = "[service-manager] "
+
+// Service interface represents pluggable Mysterium service
+type Service interface {
+	Start(providerID identity.Identity, dialogWaiter communication.DialogWaiter) (dto_discovery.ServiceProposal, error)
+	Wait() error
+	Stop() error
+}
+
+// NewManager creates new instance of pluggable services manager
+func NewManager(
+	networkDefinition metadata.NetworkDefinition,
+	identityLoader identity_selector.Loader,
+	signerFactory identity.SignerFactory,
+	identityRegistry identity_registry.IdentityRegistry,
+	service Service,
```

Comment:
here prob we should have a slice of services here. We will be creating more than 1 service soon.

Reply:
Yes, will be then needed. This is small PR to decouple and inject Openvpn service

Reply:
Why manager should get list of services at construction time? It would be nice to register/deregister any service on the fly. Especially if we are providing service management api later. I don't want to restart node simply because to enable new service

Reply:
This is why I call this PR "Inject openvpn service..". Dynamic loading will come later